File: src/easy/balanced-sums.ts
export type YesNo = "YES" | "NO";
export const balancedSums = (arr: number[]) => {
    let left = 0;
    let right = arr.reduce((accum, curr) => accum + curr, 0);
    for (const num of arr) {
        right -= num;
        if (left === right) return "YES";
        left += num;
    }
    return "NO";
};

File: easy/balanced_sums.py
from functools import reduce


def balanced_sums(arr: [int]):
    if len(arr) == 1:
        return "YES"
    left, total = 0, sum(arr)
    for num in arr:
        right = total - left - num
        if left == right:
            return "YES"
        left += num
    return "NO"
